File: src/main/java/com/shengxuan/speed/service/DeviceService.java
package com.shengxuan.speed.service;

import com.shengxuan.speed.entity.Device;
import com.shengxuan.speed.entity.pojo.*;

import java.util.List;

public interface DeviceService {
    List<Device> findAll();

    /**
     * 分页条件查询
     * @param device
     * @param pageNo
     * @param pageSize
     * @return
     */
    PageResult findPage(Device device, int pageNo, int pageSize);

    Device findById(String deviceId,int serverId);

    List<RegionAndSubRegion> findAllRegion();

    List<String> findAllDeviceType();
    List<Device> findAllAndFlag();

//    /**
//     * 查询所有区域包含子区及其设备
//     * @return
//     */
//    List<RegionAndSubRegionAndDevice> findAllRegionAndSubRegionAndDevice();

    /**
     * 查询所有server区域包含子区及其设备
     * @return
     */
    List<ServerAndRegionAndSubRegionAndDevice> findAllServerAndRegionAndSubRegionAndDevice();

}

File: src/main/resources/static/js/service/deviceService.js
app.service('deviceService',function ($http) {
    this.findAll = function () {
        return $http.get('../device/findAll');
    };
    this.findAllAndFlag = function () {
        return $http.get('../device/findAllAndFlag');
    };
    this.search = function (pageNo,pageSize,searchDevice) {
        return $http.post("../device/search?pageNo="+pageNo+"&pageSize="+pageSize,searchDevice);
    };
    this.findById = function (deviceId,serverId){
        return $http.get('../device/findById?deviceId='+deviceId+'&serverId='+serverId);
    };
    this.findAllDeviceType = function (){
        return $http.get('../device/findAllDeviceType');
    };
});